app/background-process/networks/dat/dat.js

```diff
@@ -27,29 +27,60 @@ import getFolderSize from 'get-folder-size'
 // constants
 // =
 
-import { DAT_MANIFEST_FILENAME } from '../../../lib/const'
+import { 
+  DAT_MANIFEST_FILENAME,
+  DAT_DOMAIN,
+  DAT_DEFAULT_DISCOVERY,
+  DAT_DEFAULT_BOOTSTRAP
+} from '../../../lib/const'
+
 
 // globals
 // =
 
 var drive // hyperdrive instance
+var swarm // discovery swarm instance
 var archives = {} // memory cache of archive objects. key -> archive
-var swarms = {} // memory cache of archive swarms. key -> swarm
+var archivesByDiscoveryKey = {} // mirror of the above cache, but discoveryKey -> archive
 var archivesEvents = new EventEmitter()
 
 // exported API
 // =
 
 export function setup () {
   drive = hyperdrive(archivesDb.getLevelInstance())
+  swarm = discoverySwarm({
+    hash: false,
+    utp: true,
+    tcp: true,
+    dns: {server: DAT_DEFAULT_DISCOVERY, domain: DAT_DOMAIN},
+    dht: {bootstrap: DAT_DEFAULT_BOOTSTRAP},
+    stream: function (info) {
+      // look up the archive by
+      var archive = info.channel ? archivesByDiscoveryKey[bufToStr(info.channel)] : null
+      if (!archive) {
+        // we dont yet know which feed they want, but hypercore has a protocol for asking
+        // TODO how do we configure upload?
+        debug(`${info.type} connection from ${info.host}, desired archives not yet known`)
+        return drive.replicate()
+      }
+      debug(`${info.type} connection from ${info.host} to fetch ${bufToStr(archive.key)}`)
+      return archive.replicate({
+        download: true,
+        upload: (archive.userSettings && archive.userSettings.uploadClaims && archive.userSettings.uploadClaims.length > 0)
+        // TODO fuck this^ (see dat 2.5 plans... in my notebook on the coffee table)
+      })
+    }
+  })
+  swarm.listen(3282)
```

Comment:
Yeah there could be other errors. 

I've been using once instead of on: `swarm.once('error')`, so if another error is happening it gets picked up after the second try. But let me know if you find anything, would be good to know for sure!